server/src/main/java/org/bonitasoft/console/common/server/i18n/I18n.java

```diff
@@ -5,32 +5,46 @@
  * it under the terms of the GNU General Public License as published by
  * the Free Software Foundation, either version 2.0 of the License, or
  * (at your option) any later version.
- *
+ * <p>
  * This program is distributed in the hope that it will be useful,
  * but WITHOUT ANY WARRANTY; without even the implied warranty of
  * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
  * GNU General Public License for more details.
- *
+ * <p>
  * You should have received a copy of the GNU General Public License
  * along with this program. If not, see <http://www.gnu.org/licenses/>.
  */
 package org.bonitasoft.console.common.server.i18n;
 
-import org.bonitasoft.web.toolkit.client.common.i18n.AbstractI18n;
-import org.bonitasoft.web.toolkit.client.common.texttemplate.Arg;
-
 import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
 import java.util.LinkedHashMap;
 import java.util.List;
 import java.util.Map;
 import java.util.TreeMap;
 
+import org.bonitasoft.web.toolkit.client.common.i18n.AbstractI18n;
+import org.bonitasoft.web.toolkit.client.common.texttemplate.Arg;
+import org.springframework.core.io.Resource;
+import org.springframework.core.io.support.PathMatchingResourcePatternResolver;
+import org.springframework.core.io.support.ResourcePatternResolver;
+
 /**
- * @author Séverin Moussel, Fabio Lombardi
+ * @author Séverin Moussel
+ * @author Fabio Lombardi
+ * @author Emmanuel Duchastenier
  */
 public class I18n extends AbstractI18n {
 
-    private static final File I18N_DIR = getI18nDirectory();
+    /**
+     * Property that can be set to override the default i18n *.po files, or to provide translations for new languages.
+     * Points to a folder that contains .po files
+     */
+    public static final String I18N_CUSTOM_DIR_PROPERTY = "org.bonitasoft.i18n.folder";
```

Comment:
Do we have somewhere to put all changes we need to think about when updating the documentation?